test/sql/local/iceberg_catalog_read.test

```diff
@@ -0,0 +1,106 @@
+# name: test/sql/local/iceberg_catalog_read.test
+# description: test integration with iceberg catalog read
+# group: [iceberg]
+
+require-env ICEBERG_SERVER_AVAILABLE
+
+require iceberg
+
+require parquet
+
+require httpfs
+
+statement ok
+CREATE SECRET (
+       TYPE ICEBERG,
+       ENDPOINT 'http://127.0.0.1:8181'
+  );
+
+
+statement ok
+CREATE SECRET (
+    TYPE S3,
+    KEY_ID 'admin',
+    SECRET 'password',
+    ENDPOINT '127.0.0.1:9000',
+    URL_STYLE 'path',
+    USE_SSL 0
+  );
+
+statement ok
+ATTACH '' AS my_datalake (TYPE ICEBERG);
+
+query IIIIII
+Show all tables;
+----
+my_datalake	default	table_mor_deletes	[__]	[INTEGER]	false
+my_datalake	default	table_partitioned	[__]	[INTEGER]	false
+my_datalake	default	table_unpartitioned	[__]	[INTEGER]	false
+
+statement error
+select * from table_unpartitioned
+----
+<REGEX>:.*table_unpartitioned does not exist.*
+
+statement error
+select * from table_unpartitioned
+----
+<REGEX>:.*Did you mean.*my_datalake.default.table_unpartitioned.*
+
+query III
+select * from my_datalake.default.table_unpartitioned order by all;
+----
+2023-03-01	1	a
+2023-03-02	2	b
+2023-03-03	3	c
+2023-03-04	4	d
+2023-03-05	5	e
+2023-03-06	6	f
+2023-03-07	7	g
+2023-03-08	8	h
+2023-03-09	9	i
+2023-03-10	10	j
+2023-03-11	11	k
+2023-03-12	12	l
+
+# test deletes (see provision.py for where deletes occur)
```

Comment:
scan on a table with deletes should throw a notimplementedexception for now

Reply:
This is one of those cases where the whole table is rewritten and there are no point deletes. I'll see if I can add a case with point deletes that throws an error